hummingbot/connector/exchange/bybit/bybit_auth.py

```diff
@@ -44,39 +33,73 @@ async def ws_authenticate(self, request: WSRequest) -> WSRequest:
 
     def get_referral_code_headers(self):
         """
-        Generates authentication headers required by ByBit
+        Generates referral headers
         :return: a dictionary of auth headers
         """
         headers = {
             "referer": CONSTANTS.HBOT_BROKER_ID
         }
         return headers
 
-    def add_auth_to_params(self,
-                           params: Optional[Dict[str, Any]]):
-        timestamp = int(self.time_provider.time() * 1e3)
-        request_params = params or {}
-        request_params["timestamp"] = timestamp
-        request_params["api_key"] = self.api_key
-        request_params = self.keysort(request_params)
-        signature = self._generate_signature(params=request_params)
-        request_params["sign"] = signature
-        return request_params
-
-    def _generate_signature(self, params: Dict[str, Any]) -> str:
-        encoded_params_str = urlencode(params)
-        digest = hmac.new(self.secret_key.encode("utf8"), encoded_params_str.encode("utf8"), hashlib.sha256).hexdigest()
-        return digest
-
-    def generate_ws_authentication_message(self):
+    def add_auth_headers(self, method: str, request: Optional[Dict[str, Any]]):
+        """
+        Add authentication headers in request object
+
+        :param method: HTTP method (POST, PUT, GET)
+        :param request: The request to be configured for authenticated interaction
+
+        :return: request object updated with xauth headers
+        """
+        ts = str(int(time.time() * 10 ** 3))
+
+        headers = {}
+        headers["X-BAPI-TIMESTAMP"] = str(ts)
+        headers["X-BAPI-API-KEY"] = self.api_key
+
+        if method.value == "POST":
+            signature = self._generate_rest_signature(
+                timestamp=ts, method=method, payload=request.data)
+        else:
+            signature = self._generate_rest_signature(
+                timestamp=ts, method=method, payload=request.params)
+
+        headers["X-BAPI-SIGN"] = signature
+        headers["X-BAPI-SIGN-TYPE"] = str(2)
```

Comment:
did you check where to add the UID of hummingbot?

Reply:
This function is dedicated to setting the AUTH headers.
Referral code header (`HBOT_BROKER_ID `  constant) is added from the `get_referral_code_headers`

```py
    def get_referral_code_headers(self):
        """
        Generates referral headers
        :return: a dictionary of auth headers
        """
        headers = {
            "referer": CONSTANTS.HBOT_BROKER_ID
        }
        return headers
```